lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -516,8 +518,25 @@ internal class EvaluatingCompiler(
     private fun compileNAryStringConcat(
         argThunks: List<ThunkEnv>,
         metas: MetaContainer): ThunkEnv {
+        val validTypes = setOf(STRING, SYMBOL)
```

Comment:
unknown propagation is handled before it gets here so both `lvalue` and `rvalue` are guaranteed to not be `null/missing`

Maybe we should change to use `theExprValue.scalar.stringValue()!!` so it throws an `NPE` if the preconditions are not met, i.e. hard fail if it's not possible for Scalar to give a String.

Reply:
Just to document what we talked offline `Scalar#stringValue()` contract requires that it returns `null` when `ExprValueType.isText() == false`. 

Will change the code to fail the concat operation when `ExprValueType.isText() == false` instead of doing the `is SYMBOL or STRING` check. The effect is the same but using `isText` is more inline with Scalar interface contract